docs/intro/upgrading-to-ionic-6.md

```diff
@@ -0,0 +1,443 @@
+# Upgrading to Ionic 6
+
+Follow this guide to upgrade your Ionic 5 apps to Ionic 6.
+
+## Getting Started
+
+### Angular
+
+1. Ionic 6 supports Angular 12+. Update to the latest version of Angular by following the [Angular Update Guide](https://update.angular.io/).
+2. Update to the latest version of Ionic 6:
+
+```shell
+npm install @ionic/angular@6
+```
+
+If you are using Ionic Angular Server, be sure to update that as well:
+
+```shell
+npm install @ionic/angular@6 @ionic/angular-server@6
+```
+
+3. Remove any usage of `Config.set()`. Instead, set your config in `IonicModule.forRoot()`. See the [Angular Config Documentation](../angular/config) for more examples.
+4. Remove any usage of the `setupConfig` function previously exported from `@ionic/angular`. Set your config in `IonicModule.forRoot()` instead.
+
+### React
+
+1. Ionic 6 supports React 17+. Update to the latest version of React:
+
+```shell
+npm install react@latest react-dom@latest
+```
+
+2. Update to the latest version of Ionic 6:
+
+```shell
+npm install @ionic/react@6 @ionic/react-router@6
+```
+
+3. Update the `test` field in the `scripts` object of your `package.json` to include `transformIgnorePatterns`:
+
+```json
+"scripts": {
+  "test": "react-scripts test --transformIgnorePatterns 'node_modules/(?!(@ionic/react|@ionic/react-router|@ionic/core|@stencil/core|ionicons)/)'",
+  ...
+}
+```
+
+4. Import and call `setupIonicReact` in your `App` component file. If you are also using `setupConfig`, pass your config to `setupIonicReact` instead:
+
+**Before**
+```tsx
+// App.tsx
+
+import { setupConfig } from '@ionic/react';
+
+...
+
+setupConfig({
+  mode: 'md'
+});
+```
+
+**After**
+```tsx
+// App.tsx
+
+import { setupIonicReact } from '@ionic/react';
+
+...
+
+setupIonicReact({
+  mode: 'md'
+});
+```
+
+:::note
+Developers must import and call `setupIonicReact` even if they are not setting custom config.
+:::
```

Comment:
Is this meant to format a certain way? Doesn't look like it's working in the rendered page 🤔 (Also applies in a few other spots)

Reply:
I believe `:::note` comes from docusaurus. Is our revised docs being used with that tool? If so we can make use of a lot of micro enhancements like:

```
```tsx title="App.tsx"
```

Reply:
Should look like this 
![image](https://user-images.githubusercontent.com/2721089/144906209-76bc2c6b-988c-4da3-b915-27deb6670cc8.png)

Reply:
Note: You will need to preview it on the test docs site: https://ionic-docs-git-6-migration-ionic1.vercel.app/docs/intro/upgrading-to-ionic-6

Reply:
Yeah it's using docusaurus